scheduler/src/cook/compute_cluster.clj

```diff
@@ -64,10 +86,11 @@
 (defn safe-kill-task
   "A safe version of kill task that never throws. This reduces the risk that errors in one compute cluster propagate and cause problems in another compute cluster."
   [{:keys [name] :as compute-cluster} task-id]
-  (try
-    (kill-task compute-cluster task-id)
-    (catch Throwable t
-      (log/error t "In compute cluster" name ", error killing task" task-id))))
+  (locking kill-lock-object
```

Comment:
Why not put the `locking` inside the `try`?

Reply:
It can go either way in the current code. However, this is safer; as I expect us to add additional logic to the catch handler per some queued tickets.

Reply:
OK, sounds good